### PR TITLE
Enhancement: Allow injecting environment

### PR DESCRIPTION
**Pull request by localheinz**
This PR

* [x] allows injecting an instance of `Env` when constructing a `CiDetector`

Related to https://github.com/infection/infection/pull/939.

💁‍♂ This would make it possible to use it `CiDetector` as a collaborator, without the need to set up environment variables for tests.

### CODE REVIEW

**Review by sanmai — APPROVED** [review]

**Comment by localheinz** [comment]
@OndraM 

Before merging, perhaps you might be interested in https://github.com/ergebnis/environment-variables? Still working on it, but perhaps you find it useful.

**Review by drupol — APPROVED** [review]
:+1:

File: src/CiDetector.php
Comment (OndraM) [inline src/CiDetector.php]: I don't like to pollute the default constructor with this optional parameter - which is something you'd probably only use in tests.

I'm thinking about using named constructor for these cases, like this:

```php
    public function __construct()
    {
        $this->environment = new Env();
    }

    public function createFromEnvironment(Env $environment)
    {
        $ciDetector = new self();
        $ciDetector->environment = $environment;

        return $ciDetector;
    }
```

What do you think?

File: src/CiDetector.php
Comment (OndraM) [inline src/CiDetector.php]: Single-ling phpdoc for `@var`, please

**Review by OndraM — CHANGES_REQUESTED** [review]

File: src/CiDetector.php
Comment (localheinz) [reply src/CiDetector.php]: Adjusted accordingly!

File: src/CiDetector.php
Comment (localheinz) [reply src/CiDetector.php]: Adjusted - also see #58.

**Review by OndraM — APPROVED** [review]

**Comment by localheinz** [comment]
@OndraM 

Is there anything that I can do to help move forward here, especially, a new release?

**Comment by OndraM** [comment]
@localheinz Nothing, it was just stuck in my backlog. I made just cosmetic changes in test names, will merge after CI. Thanks! :1st_place_medal:

**Comment by localheinz** [comment]
Thank you, @OndraM!

**Comment by OndraM** [comment]
Also, this has just been released as part of version 3.3.0 :ok_hand:

**Pull request merged**
